cimgen/languages/modernpython/lang_pack.py

```diff
@@ -81,21 +85,135 @@ def _get_type_and_default(text, renderer) -> tuple[str, str]:
+    if datatype.lower() == "datetime":
+        return "datetime"
+    if datatype.lower() == "monthday":
+        return "str"  # TO BE FIXED? I could not find a datatype in python that holds only month and day.
+    if datatype.lower() == "date":
+        return "date"
+    # as of today no CIM model is using only time.
+    if datatype.lower() == "time":
+        return "time"
+    if datatype.lower() == "float":
+        return "float"
+    if datatype.lower() == "string":
+        return "str"
+    else:
+        # this actually never happens
+        return "float"
+
+
+def _set_imports(attributes):
+    classes = set()
+    for attribute in attributes:
+        if attribute["is_cim_datatype"] or attribute["is_primitive_attribute"]:
+            classes.add(attribute["attribute_class"])
+
+    result = ""
+    for val in classes:
+        result += "from ." + val + " import " + val + "\n"
+    return result + "\n"
+
+
+def _compute_cim_data_type(attributes) -> dict:
+    cim_attributes = {}
+    cim_attributes["python_type"] = "None"
+    cim_attributes["unit"] = "UnitSymbol.none"
+    cim_attributes["multiplier"] = "UnitMultiplier.none"
+
+    for attribute in attributes:
+        if (
+            "about" in attribute
+            and attribute["about"]
+            and "value" in attribute["about"]
+            and "attribute_class" in attribute
+        ):
+            cim_attributes["python_type"] = _primitive_to_data_type(attribute["attribute_class"])
+        if (
+            "about" in attribute
+            and attribute["about"]
+            and "multiplier" in attribute["about"]
+            and "isFixed" in attribute
+        ):
+            cim_attributes["multiplier"] = "UnitMultiplier." + attribute["isFixed"]
+        if "about" in attribute and attribute["about"] and "unit" in attribute["about"] and "isFixed" in attribute:
+            cim_attributes["unit"] = "UnitSymbol." + attribute["isFixed"]
+    return cim_attributes
+
+
+def _set_cim_data_type(text, render) -> str:
+    attribute = eval(render(text))
+    cim_data_type = ""
+    if attribute["is_cim_datatype"] or attribute["is_primitive_attribute"]:
+        cim_data_type += "\n            "
+        cim_data_type += """"cim_data_type": """ + attribute["attribute_class"] + ","
+    return cim_data_type
+
+
+# called by chevron, text contains the label {{dataType}}, which is evaluated by the renderer (see class template)
+def _set_instances(text, render):
+    instance = None
+    try:
+        # render(text) returns a python dict.
+        # Some fields might be quoted by '&quot;' instead of '"', making the first eval fail.
+        instance = ast.literal_eval(render(text))
+    except SyntaxError as se:
+        rendered = render(text)
+        rendered = rendered.replace("&quot;", '"')
+        instance = eval(rendered)
+        logger.warning("Exception in evaluating %s : %s . Handled replacing quotes", rendered, se.msg)
+    if "label" in instance:
```

Comment:
Wouldn't it be easier to replace before the exception?
```suggestion
def _set_instances(text, render):
    # render(text) returns a python dict as string.
    # Some fileds might be quoted by '"' instead of '"'.
    instance = ast.literal_eval(render(text).replace(""", '"'))
    if "label" in instance:
```